invite_beginner_tutorials/src/objects.cpp

```diff
@@ -0,0 +1,112 @@
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sachin Chitta, Dave Coleman 
+   Modified by: Daniel Ordonez Apraez - daniels.ordonez@gmail.com
+/* 
+This document is a modification of the Move Grup Moveit Tutorial, in order 
+for it to work with CSDA10F robot, it is intended to familiarize the operator with several 
+Moveit classes that are commonly used, and most of the "common robot 
+operations".
+
+We formaly thank SwRI for ofering this code under the BSD licence and as 
+requested by the software license agreement we state that we do not have 
+any relationship with SwRI and that they do not hold any responsability whatsoever over
+this code.
+
+*/ 
+
+//For add object
+#include "geometric_shapes/shapes.h"
+#include "geometric_shapes/mesh_operations.h"
+#include "geometric_shapes/shape_operations.h"
+#include <moveit/move_group_interface/move_group.h>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+#include <geometric_shapes/shape_operations.h>
+using namespace Eigen;
+//********
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "move_group_interface_tutorial");
+  ros::NodeHandle node_handle;
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
+
+  // Define a collision object ROS message.
+
+   moveit::planning_interface::PlanningSceneInterface current_scene;
+   Vector3d b(0.001, 0.001, 0.001); // vector para escalar
+   moveit_msgs::CollisionObject co;
+
+
+  // The id of the object is used to identify it.
+   co.id = "glass";
+
+    
+    shapes::Mesh* m = shapes::createMeshFromResource("package://invite_beginner_tutorials/meshes/glass.stl",b); 
+    ROS_INFO("Glass mesh loaded");
+
+    shape_msgs::Mesh mesh;
+    shapes::ShapeMsg mesh_msg;  
+    shapes::constructMsgFromShape(m, mesh_msg);
+    mesh = boost::get<shape_msgs::Mesh>(mesh_msg);
+
+    co.meshes.resize(1);
+    co.mesh_poses.resize(1);
+    co.meshes[0] = mesh;
+    co.header.frame_id = "glass";
+
```

Comment:
What are this parameters how can they be used to edit other models? which coordinate frame would be the parent or reference frame for the pose coordinates ? "glass"?